src/commands/mod.rs

```diff
@@ -2054,10 +2031,11 @@ Adjust the right side until the diff shows the changes you want to move
 to the destination. If you don't make any changes, then all the changes
 from the source will be moved into the destination.
 ",
-        workspace_command.format_commit_summary(&source),
-        workspace_command.format_commit_summary(&destination)
+        tx.base_workspace_helper().format_commit_summary(&source),
+        tx.base_workspace_helper()
+            .format_commit_summary(&destination)
```

Comment:
It feels like we could remove the calls to `base_workspace_helper()` by creating the instructions be we start the transaction.

I think `select_diff()`, on the other hand, should be moved onto `WorkspaceCommandTransaction` since it creates new objects in the backend (I'm thinking about #50).

Just a thought. Feel free to leave it all as is.

Reply:
> we could remove the calls to base_workspace_helper() by creating the instructions be we start the transaction.

Yeah, that's doable, but I decided to not because `cmd_unsquash()` has `.edit_diff()` in `if interactive` branch. Since caller wouldn't have to care about mut_repo vs base_repo here, it's probably simpler to add `tx.format_commit_summary()` that uses mut_repo.

> I think select_diff(), on the other hand, should be moved onto WorkspaceCommandTransaction since it creates new objects in the backend (I'm thinking about https://github.com/martinvonz/jj/issues/50).

Good point. I was wondering if tree creation should be covered by transaction. I'll move `select_diff()`/`edit_diff()` to the transaction wrapper.

Reply:
Split to #1122